backend/bitboxbase/rpcclient/websocket.go

```diff
@@ -72,23 +74,31 @@ func (rpcClient *RPCClient) runWebsocket(client *websocket.Conn, writeChan <-cha
 	writeLoop := func() {
 		defer func() {
 			_ = client.Close()
+			rpcClient.log.Info("Closing websocket write loop")
 		}()
 		for {
 			select {
-			case message, ok := <-writeChan:
-				if !ok {
-					_ = client.WriteMessage(websocket.CloseMessage, []byte{})
-					return
-				}
-				err := client.WriteMessage(websocket.BinaryMessage, rpcClient.sendCipher.Encrypt(nil, nil, message))
-				if err != nil {
-					rpcClient.log.WithFields(logrus.Fields{"group": "rpcClient websocket", "error": err}).Error("websocket could not write message")
-				}
-
-			case <-closeChan:
+			case <-rpcClient.rpcConnection.CloseChan():
 				_ = client.WriteMessage(websocket.CloseMessage, []byte{})
 				rpcClient.log.Info("closing websocket connection")
 				return
+			default:
+				select {
+				case <-rpcClient.rpcConnection.CloseChan():
+					_ = client.WriteMessage(websocket.CloseMessage, []byte{})
+					rpcClient.log.Info("closing websocket connection")
+					return
+
+				case message, ok := <-writeChan:
+					if !ok {
+						_ = client.WriteMessage(websocket.CloseMessage, []byte{})
+						return
+					}
+					err := client.WriteMessage(websocket.BinaryMessage, rpcClient.sendCipher.Encrypt(nil, nil, message))
+					if err != nil {
+						rpcClient.log.WithFields(logrus.Fields{"group": "rpcClient websocket", "error": err}).Error("websocket could not write message")
```

Comment:
"group" really needed? in any case, you can use WithError(err).WithFields(...) to not manually put the error there.